### ISSUE TITLE
Test mode VRAM requirements with cuDNN

### ISSUE THREAD

**Issue by gavinmh** [KaimingHe/deep-residual-networks#42]
Running in test mode on a GeForce GTX TITAN X (driver version 367.57), with cuDNN-4, batch size 1 and a 224x224 image requires 10740 MB of VRAM. This doesn't seem correct given that VGGNet, etc. require < 4 GB VRAM in test mode for similar batches. Does storing intermediate gradients require this much memory? Are my results sane?

**Comment by mrgloom** [comment]
With 8Gb GPU (gtx 1070 ) I can run ResNet-152 only with batch size 1, so as I understand only finetuning is feasible with such small batchsize, I also wonder what are theoretical \ practival memory requirements for family of ResNet models (ResNet-50, ResNet-101, ResNet-152) for training and for forward pass?

**Comment by gavinmh** [comment]
There was an error in my script; my batch size was 35, not 1.
